### PR TITLE
Update to SPIRE 1.9.2, use retry_bootstrap, remove wait for it

### PR DESCRIPTION
**Pull request by faisal-memon**
Uses the new `retry_bootstrap` feature in 1.9.2 so we don't need wait for it in agent anymore.

### CODE REVIEW

File: charts/spire/charts/spire-agent/templates/daemonset.yaml [hunk 1]
```diff
@@ -115,17 +115,6 @@ spec:
               mountPath: /tpm
           imagePullPolicy: {{ .Values.nodeAttestor.tpmDirect.plugin.image.pullPolicy }}
         {{- end }}
-        - name: init
-          # This is a small image with wait-for-it, choose whatever image
-          # you prefer that waits for a service to be up. This image is built
-          # from https://github.com/vishnubob/wait-for-it
-          image: {{ template "spire-lib.image" (dict "image" .Values.waitForIt.image "global" .Values.global) }}
```
Comment (kfox1111) [inline charts/spire/charts/spire-agent/templates/daemonset.yaml:122]: The pr is looking pretty good.

I think its leaving behind a 'waitForIt' section in values.yaml. Could you remove that too please?

Comment (faisal-memon) [reply charts/spire/charts/spire-agent/templates/daemonset.yaml:122]: Removed

Comment (kfox1111) [reply charts/spire/charts/spire-agent/templates/daemonset.yaml:122]: Oh. Found one more reference:
https://github.com/spiffe/helm-charts-hardened/blob/main/.github/tests/images.json#L10-L14

Comment (faisal-memon) [reply charts/spire/charts/spire-agent/templates/daemonset.yaml:122]: Thanks, fixed

**Review by kfox1111 — APPROVED** [review]
LGTM. Thanks! :)

**Pull request merged**